File: runtime/python/units/wasi/fd_advise.py
def wasi_fd_advise(self, fd, offset, length, advice):
    io = self.fds.get(fd)
    if io is None or isinstance(io, self.WasiDir):
        return self.ERRNO_BADF
    # posix_fadvise is purely advisory; validating the fd and succeeding is a
    # faithful no-op.
    return self.ERRNO_SUCCESS

File: runtime/perl/units/wasi/fd_advise.pl
# requires:
sub wasi_fd_advise {
    my ($self, $fd, $offset, $len, $advice) = @_;
    my $e = $self->{fds}{$fd};
    return ERRNO_BADF if !defined($e) || $e->{dir};
    # posix_fadvise is purely advisory; validating the fd and succeeding is
    # a faithful no-op.
    return ERRNO_SUCCESS;
}
